python/perf-kernels/flash-attention.py

```diff
@@ -617,10 +612,29 @@ def _attn_bwd_preprocess(
     else:
         tl.store(delta_ptrs, delta)
 
+@triton.jit
+def print_gpu(prefix, val=None):
+    if (tl.program_id(0) == 0) and ((tl.program_id(1) == 0) and (tl.program_id(2) == 0)):
+        if val is not None:
+            tl.device_print(prefix, val)
+        else:
+            tl.device_print(prefix)
+
+@triton.jit
+def compute_alibi_block(alibi_slope, seqlen_k, seqlen_q, offs_m, offs_n, transpose = False):
+    # compute the relative position using the global positions
+    if transpose:
+        relative_pos_block = offs_n[:,None] + seqlen_q - offs_m[None,:] - seqlen_k
```

Comment:
If possible, may be add a comment with the intuition on why it goes from offs_m + seqlen_k to offs_m - seqlen_k when transposing. IOW, flipping the dims is intuitive as it is transpose, the change of sign is not.

Reply:
I just transposed the normal computation. It is just changing a view of the block so I don't think it will be that expensive. I also added notes working through an example.